ScoutSuite/providers/azure/authentication_strategy.py

```diff
@@ -40,6 +40,13 @@ def authenticate(self,
             logging.getLogger('msrest').setLevel(logging.ERROR)
             logging.getLogger('urllib3').setLevel(logging.ERROR)
 
+            cloud_dic = {"AZURE_PUBLIC_CLOUD":AZURE_PUBLIC_CLOUD,
+                         "AZURE_CHINA_CLOUD":AZURE_CHINA_CLOUD,
+                         "AZURE_US_GOV_CLOUD":AZURE_US_GOV_CLOUD,
+                         "AZURE_GERMAN_CLOUD":AZURE_GERMAN_CLOUD}
+
+            cloud_env = cloud_dic.get(kargs.get('azure_env'), AZURE_PUBLIC_CLOUD)
+
             if cli:
```

Comment:
No changes are needed for this to work with CLI authentication?

Reply:
We have a tight environment that won't let us use Azure CLI. We connect via service principal. The snag in our case is in the call to the ServicePrincipalCredentials.

Reply:
Looks like it's working for all facades but graphrbac, I'm getting a number of errors:
```
2019-10-18 09:46:54 wrkbx scout[17579] ERROR graphrbac.py L22: Failed to retrieve users: Operation returned an invalid status code 'Not Found'
2019-10-18 09:46:54 wrkbx scout[17579] ERROR graphrbac.py L29: Failed to retrieve groups: Operation returned an invalid status code 'Not Found'
2019-10-18 09:46:54 wrkbx scout[17579] ERROR graphrbac.py L46: Failed to retrieve service principals: Operation returned an invalid status code 'Not Found'
2019-10-18 09:46:54 wrkbx scout[17579] ERROR graphrbac.py L53: Failed to retrieve applications: Operation returned an invalid status code 'Not Found'
```
These disappear if I remove the `base_url` parameter.

Reply:
Thanks for checking. I was able to reproduce. Made the following changes but was not quite able to get pass through it due to "Insufficient privileges" return. I have given most read access under Azure Active Directory and Microsoft Graph but I still get the same message.

The changes I did is:
        self._client = GraphRbacManagementClient(graphrbac_credentials, tenant_id=tenant_id, base_url=graphrbac_credentials.cloud_environment.endpoints.active_directory_graph_resource_id)

so basically using active_directory_graph_resource_id instead of  resource_manager